arithmetization/src/main/java/net/consensys/linea/rpc/linea/LineaSetExtraData.java

```diff
@@ -0,0 +1,124 @@
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.extern.slf4j.Slf4j;
+import net.consensys.linea.extradata.LineaExtraDataException;
+import net.consensys.linea.extradata.LineaExtraDataHandler;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import org.hyperledger.besu.datatypes.rpc.JsonRpcResponseType;
+import org.hyperledger.besu.datatypes.rpc.RpcMethodError;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.parameters.JsonRpcParameter;
+import org.hyperledger.besu.plugin.services.RpcEndpointService;
+import org.hyperledger.besu.plugin.services.exception.PluginRpcEndpointException;
+import org.hyperledger.besu.plugin.services.rpc.PluginRpcRequest;
+
+@Slf4j
+public class LineaSetExtraData {
+
+  private static final AtomicInteger LOG_SEQUENCE = new AtomicInteger();
+  private final JsonRpcParameter parameterParser = new JsonRpcParameter();
+  private final RpcEndpointService rpcEndpointService;
+  private LineaExtraDataHandler extraDataHandler;
+
+  public LineaSetExtraData(final RpcEndpointService rpcEndpointService) {
+    this.rpcEndpointService = rpcEndpointService;
+  }
+
+  public void init(final LineaExtraDataHandler extraDataHandler) {
+    this.extraDataHandler = extraDataHandler;
+  }
+
+  public String getNamespace() {
+    return "linea";
+  }
+
+  public String getName() {
+    return "setExtraData";
+  }
+
+  public Boolean execute(final PluginRpcRequest request) {
+    if (log.isDebugEnabled()) {
+      // no matter if it overflows, since it is only used to correlate logs for this request,
+      // so we only print callParameters once at the beginning, and we can reference them using the
+      // sequence.
+      LOG_SEQUENCE.incrementAndGet();
+    }
+
+    final var extraData = parseRequest(request.getParams());
+
+    try {
+      updatePricingConf(extraData);
+
+      updateStandardExtraData(extraData);
+
+      return Boolean.TRUE;
+    } catch (final LineaExtraDataException lede) {
+      throw new PluginRpcEndpointException(new ExtraDataPricingError(lede));
+    }
+  }
+
+  private void updateStandardExtraData(final Bytes32 extraData) {
+    final var resp =
+        rpcEndpointService.call("miner_setExtraData", new Object[] {extraData.toHexString()});
+    if (!resp.getType().equals(JsonRpcResponseType.SUCCESS)) {
+      throw new LineaExtraDataException(
+          LineaExtraDataException.ErrorType.FAILED_CALLING_SET_EXTRA_DATA,
+          "miner_setExtraData failed: " + resp);
+    }
+  }
+
+  private void updatePricingConf(final Bytes32 extraData) {
+    extraDataHandler.handle(extraData);
+    log.atDebug()
+        .setMessage("[{}] Successfully handled extra data pricing")
+        .addArgument(LOG_SEQUENCE::get)
+        .log();
+  }
+
+  private Bytes32 parseRequest(final Object[] params) {
+    final var rawParam = parameterParser.required(params, 0, String.class);
+    final var extraData = Bytes32.wrap(Bytes.fromHexStringLenient(rawParam));
```

Comment:
what will be the behavior is we pass more than 32 bytes ? I tried to see but I don't think there is a test for that

Reply:
implemented a test for that, and found that instead of invalid params it was returning internal error, so I fixed it, thanks for pointing me to it